ui/src/App.tsx

```diff
@@ -375,6 +383,105 @@ const defaultImmunizations: Promise<StoredSHC[]> = Promise.all(
   }),
 );
 
+export function SHLinkCreateCustom() {
+  let navigate = useNavigate();
+  let { store, dispatch } = useStore();
+  // display all vaccines 
+  let allVaccineCards = Object.values(store.vaccines);
+  let [usePin, setUsePin]= useState(false);
+  let [pin, setPin] = useState('1234');
+  let [expires, setExpires] = useState(false);
+  // list of checked vaccination ids to include in the created SHLink
+  const [checkedVaccinations, setCheckedVaccinations] = useState<number[]>([]);
```

Comment:
Would consider reducing the number of "moving parts" in state here by tracking just an array of booleans (each mapped to a checkbox). Then you can derive the rest, like:

~~const [checkedVaccinations, setCheckedVaccinations] = useState<number[]>([]);~~

```ts
const checkedVaccinations = allVaccineCards[i]
  .map(card => card.id)
  .filter((card, i) => checkedVaccinations[i] === true);

const handleOnChange = (index: Number) => {
    const updatedCheck = isChecked.map((item: boolean, i: Number) => {
      return i === index ? !item : item
    });
    setIsChecked(updatedCheck);
  }
```

Reply:
Thank you for the feedback! I agree with your note about reusing the same path when creating a custom dataset. I pushed some changes to address this, and I also submitted a new PR to the upstream repository if you would like to leave more comments there. Thanks!